src/commands.cpp

```diff
@@ -2351,36 +2351,95 @@ void CommandEnumerateQuotaItems::procresult()
-                || ((gbstorage = int(client->json.getint())) < 0)
-                || ((gbtransfer = int(client->json.getint())) < 0)
-                || ((months = int(client->json.getint())) < 0)
-                || !(a = client->json.getvalue())
-                || !(c = client->json.getvalue())
-                || !(d = client->json.getvalue())
-                || !(ios = client->json.getvalue())
-                || !(android = client->json.getvalue()))
+        bool finished = false;
+        while (!finished)
         {
-            return client->app->enumeratequotaitems_result(API_EINTERNAL);
+            switch (client->json.getnameid())
+            {
+                case MAKENAMEID2('i', 't'):
+                    type = int(client->json.getint());
+                    break;
+                case MAKENAMEID2('i', 'd'):
+                    product = client->json.gethandle(8);
+                    break;
+                case MAKENAMEID2('a', 'l'):
+                    prolevel = int(client->json.getint());
+                    break;
+                case 's':
+                    gbstorage = int(client->json.getint());
+                    break;
+                case 't':
+                    gbtransfer = int(client->json.getint());
+                    break;
+                case 'm':
+                    months = int(client->json.getint());
+                    break;
+                case 'p':
+                    a = client->json.getvalue();
+                    break;
+                case 'c':
+                    c = client->json.getvalue();
+                    break;
+                case 'd':
+                    d = client->json.getvalue();
+                    break;
+                case MAKENAMEID3('i', 'o', 's'):
+                    ios = client->json.getvalue();
+                    break;
+                case MAKENAMEID6('g', 'o', 'o', 'g', 'l', 'e'):
+                    android = client->json.getvalue();
+                    break;
+                case MAKENAMEID3('m', 'b', 'p'):
+                    m = client->json.getvalue();
+                    break;
+                case EOO:
+                    if (type < 0
+                            || ISUNDEF(product)
+                            || (prolevel < 0)
+                            || (!type && gbstorage < 0)
+                            || (!type && gbtransfer < 0)
+                            || (months < 0)
+                            || !a
+                            || !c
+                            || !d
+                            || !m
+                            || !ios
+                            || !android)
+                    {
+                        return client->app->enumeratequotaitems_result(API_EINTERNAL);
+                    }
+
+                    // If pricing plan is business, storage and bandwith are unlimited
+                    if (type == 1)
+                    {
+                        gbstorage = 0;
```

Comment:
I think we should honor the value returned by API: `-1` to indicate the storage/bandwidth is unlimited (which is different than `0` --> no allowance at all).
If we do the change above, we need to change the public interfaces of `MegaPricing` for `getGbStorage()` and `getGbBandwidth()`, but I feel the subtle difference between `-1` and `0` is worthy.